R/FilterUnivariateCox.R

```diff
@@ -0,0 +1,65 @@
+#' @title Univariate Cox Survival Filter
+#'
+#' @name mlr_filters_univariate_cox
+#'
+#' @description Calculates scores for assessing the relationship between
+#' individual features and the time-to-event outcome (right-censored survival
+#' data) using a univariate Cox proportional hazards model.
+#' The goal is to determine which features have a statistically significant
+#' association with the event of interest, typically in the context of clinical
+#' or biomedical research.
+#'
+#' This filter fits a [Cox Proportional Hazards][survival::coxph()] model using
+#' each feature independently and extracts the \eqn{p}-value that quantifies the
+#' significance of the feature's impact on survival. The filter value is
+#' `-log10(p)` where `p` is the \eqn{p}-value. This transformation is necessary
+#' to ensure numerical stability for very small \eqn{p}-values. Also higher
+#' values denote more important features. The filter works only for numeric
+#' features so please ensure that factor variables are properly encoded, e.g.
+#' using [PipeOpEncode][mlr3pipelines::PipeOpEncode].
+#'
+#' @family Filter
+#' @include Filter.R
+#' @template seealso_filter
+#' @export
+#' @examples
+#'
+#' filter = flt("univariate_cox")
+#' filter
+#'
+FilterUnivariateCox = R6Class("FilterUnivariateCox",
+  inherit = Filter,
+  public = list(
+    #' @description Create a FilterUnivariateCox object.
+    initialize = function() {
+      super$initialize(
+        id = "surv.univariate_cox",
+        packages = "survival",
+        param_set = ps(),
+        feature_types = c("integer", "numeric", "logical"),
+        task_types = "surv",
+        label = "Univariate Cox Survival Score",
+        man = "mlr3filters::mlr_filters_univariate_cox"
+      )
+    }
+  ),
+
+  private = list(
+    .calculate = function(task, nfeat) {
+      t = task$clone()
+      features = t$feature_names
+
+      scores = map_dbl(features, function(feature) {
+        t$col_roles$feature = feature
+        model = invoke(survival::coxph, formula = t$formula(), data = t$data())
```

Comment:
This queries the target columns again in each iteration.
Please query the target columns once and cbind them in each iteration.

Reply:
I don't understand, where are the target columns queried? I followed the same thing done in this filter => https://github.com/mlr-org/mlr3filters/blob/main/R/FilterPerformance.R#L81

Reply:
Yeah, FilterPerformance is also suboptimal but there we are calling into the `$train()` API of the learner, which does not allow passing a `data.frame` but expects a task (this is a shortcoming of our design). Because you are passing a `data.frame` into `coxph()`, you can optimize this.  The target columns are queried in the `task$data()` call. In most cases (depending on the backend) this does a subset on a `data.table` using the target columns and the feature columns. Because the target column is the same in each map iteration, there is room to optimize